src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceReader.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+
+@NoArgsConstructor
+public class BasicGremlinSourceReader {
+
+    <T> T createDomainInstance(@NonNull Class<T> type) {
+        final T domain;
+
+        try {
+             domain = type.newInstance();
+        } catch (IllegalAccessException e) {
+            throw new IllegalArgumentException("can not access type constructor", e);
+        } catch (InstantiationException e) {
+            throw new IllegalArgumentException("failed to create instance of given type", e);
+        }
+
+        return domain;
```

Comment:
Do not understand why this `createDomainInstance` is a method in BasicGremlinSourceReader, I think the function here is an existing feature/function of Java lang, the only thing we do here is wrapped the exception to Runtime exception, can we make it a util class if what we do here is only wrapping exception.

Reply:
Make sense, should be put it into utils.